backstory/src/main/java/com/google/sps/story/StoryManager.java

```diff
@@ -30,4 +30,14 @@
    * @param factory StoryManagerRequestFactory to use for HttpRequests.
    */
   public void setRequestFactory(StoryManagerRequestFactory factory);
+
+  /**
+   * Cycles selectedUrlIndex to a backup url.
+   */
+  public void cycleUrl();
+
+  /**
+   * Sets prompt for text generation.
+   */
+  public void setPrefix(String prefix);
```

Comment:
This looks unused in this PR... what will it be used for?

Reply:
The backstory servlet will likely be re-using the storyManager object and cycling the URL, so this lets it change the prompt on each request without re-instantiating the object.

Reply:
I see... I think making `prefix` mutable could introduce unexpected behavior, not with the current usage of this class, but consider this use case:

1. FooService is a user of StoryManager.
2. BarService is also a user of StoryManager.
3. BazService owns both FooService and BarService. When initializing FooService and BarService, BazService creates a new instance of StoryManager with prompt X, and uses that StoryManager to construct FooService and BarService.
4. Now FooService and BarService share the same StoryManager.
5. FooService decides at some point to change the prompt to Y.
6. BarService continues to use StoryManager assuming that the prompt is still X.

Perhaps a bit contrived, but I think it would be simpler if StoryManager was not mutable like this... it avoids the need to consider edge cases that may occur if the prompt is changed.

One solution may be to create a URLProvider:

```
class StoryManagerURLProvider {
  String getCurrentURL();
  void cycleURL();
}

class StoryManagerImpl {
  // The constructor accepts an instance of the URL Provider.
  StoryManagerImpl(StoryManagerURLProvider urlProvider, ...);

  boolean cycleURL() {
    urlProvider.cycleURL();
  }
}
```

Then you could have multiple StoryManager instances sharing the same URLProvider, and not need to worry about that type of mutation.

You would also probably want to make StoryManagerURLProvider thread-safe:
```
class StoryManagerURLProvider {
  synchronized String getCurrentURL();
  synchronized void cycleURL();
}
```

In case text generation calls become parallel in the future.

What do you think?

Reply:
@UdayKalra ping

Reply:
Yeah, I agree. I'll set this up and PR it soon.